src/aeb_fate_data.erl

```diff
@@ -133,30 +133,25 @@ format(?FATE_TRUE) -> "true";
 format(?FATE_FALSE) -> "false";
 format(?FATE_NIL) -> "[]";
 format(L) when ?IS_FATE_LIST(L) -> format_list(?FATE_LIST_VALUE(L));
-format(?FATE_UNIT) -> "unit";
+format(?FATE_UNIT) -> "()";
 format(?FATE_TUPLE(T)) ->
-    "{ " ++ [format(E) ++ " " || E <- erlang:tuple_to_list(T)] ++ "}";
+    ["( ", lists:join(", ", [ format(E) || E <- erlang:tuple_to_list(T)]), " )"];
 format(S) when ?IS_FATE_STRING(S) -> [S];
 format(?FATE_VARIANT(Size, Tag, T)) ->
-    "( " ++ integer_to_list(Size) ++ ", "
-        ++ integer_to_list(Tag) ++ ", "
-        ++ [format(E) ++ " " || E <- erlang:tuple_to_list(T)]
-        ++ " )";
+    ["(| ",
+      lists:join("| ", [integer_to_list(Size), integer_to_list(Tag) |
+                        [format(E) || E <- erlang:tuple_to_list(T)]]),
+     " |)"];
 format(M) when ?IS_FATE_MAP(M) ->
-    "#{ "
-        ++ format_kvs(maps:to_list(?FATE_MAP_VALUE(M)))
-        ++" }";
-format(?FATE_ADDRESS(Address)) -> address_to_base58(Address);
+    ["{ ", format_kvs(maps:to_list(?FATE_MAP_VALUE(M))), " }"];
+format(?FATE_ADDRESS(Address)) -> ["#", address_to_base58(Address)];
```

Comment:
seems not yet to handle "bits"

Reply:
They will turn into an int for now... Ill fix.